main.tf

```diff
@@ -1,3 +1,9 @@
-/********************************************************************
-This file is used to implement the ROOT module.
-*********************************************************************/
+resource "ibm_resource_instance" "ease_instance" {
+  name              = var.ease_name
+  resource_group_id = var.resource_group_id
+  service           = "ease"
+  plan              = "free"
+  location          = "us-east"
+  tags              = var.tags
+  parameters        = var.parameters == {} ? null : { "sourceRepoURL" : var.parameters.sourceRepoURL, "configRepoURL" : var.parameters.configRepoURL }
```

Comment:
Since you are building the map here as well, I would consider my comment in the variables to just have the two URL inputs as separate inputs, and not part of a "parameters" string. The only change here needed for that is the first check to see if the entire map would be null or not.

If the resulting IF condition starts looking ugly on this one line, you could consider a boolean local, like "add_parameters" or similar, and check for either source or config URL != null for true, then use that local on the front end here to set null vs build the map.

Reply:
This change has already been made.